core/src/test/scala/unit/kafka/raft/RaftManagerTest.scala

```diff
@@ -198,7 +200,8 @@ class RaftManagerTest {
   @Test
   def testShutdownIoThread(): Unit = {
     val raftClient = mock(classOf[KafkaRaftClient[String]])
-    val ioThread = new RaftIoThread(raftClient, threadNamePrefix = "test-raft")
+    val faultHandler = mock(classOf[FaultHandler])
```

Comment:
use `MockFaultHandler` here

then, lalter in the test, do `maybeRethrowFirstException` to ensure no faults happened